### ISSUE TITLE
Triangle position is broken

### ISSUE THREAD

**Issue by HymanZHAN** [Hacker0x01/react-datepicker#3176]
**Describe the bug**
The placement of the triangle looks broken.

**To Reproduce**
Use the date picker component in a place where the `react-datepicker__input-container` is wide.

Go to the project's homepage and the one in the hero section is a good example.

**Expected behavior**
The triangle should have a right margin of some sort to prevent the broken look.

**Screenshots**
![image](https://user-images.githubusercontent.com/26806995/132284000-8ddd6c4d-64c8-419d-989a-57f778927141.png)

![image](https://user-images.githubusercontent.com/26806995/132284039-c74281cb-6559-49eb-80a1-95139a918e3c.png)

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Firefox, Edge (Chromium)
- Version: FF 91, Edge 93

**Additional context**
N/A

**Comment by mhatvan** [comment]
Encountering this as well in Storybook with a non-wide input container.

**Comment by mhatvan** [comment]
Importing `react-datepicker.min.css` instead of `react-datepicker.css` fixes the triangle issue as a workaround.

**Comment by brightertools** [comment]
> Importing `react-datepicker.min.css` instead of `react-datepicker.css` fixes the triangle issue as a workaround.

This didn't fix it for me, also seeing this triangle issue.. and using bootstrap, not sure if that contributes to the issue, if I remove the form-control class there is still an issue.

![datepickerissue](https://user-images.githubusercontent.com/6533238/135574150-99c8ed5b-07b3-4b10-9f75-2692714fff9a.png)

overriding with this fixed it for me:

```
.react-datepicker__triangle {
    left: -50px !important;
}
```

**Comment by kingston** [comment]
I dug into the code a bit and realized the arrow position is actually determined by Popper and you can modify the Popper arrow position via the popperModifiers property on the DatePicker component (https://popper.js.org/docs/v2/modifiers/arrow/).

The following code snippet (property on DatePicker) will make sure the arrow doesn't move too far to the right by adding 24px of padding to the arrow position:
```
      popperModifiers={[
        {
          name: 'arrow',
          options: { padding: 24 },
        },
      ]}
```

<img width="421" alt="image" src="https://user-images.githubusercontent.com/339308/193011055-d1aca354-9713-466f-899b-b960df4a0a60.png">

I didn't like how it being to the right so the following code snippet will make the arrow stay to the left of the input always: (haven't tested too thoroughly but seems to work well)
```
      popperModifiers={[
        {
          name: 'arrow',
          options: {
            padding: ({ popper, reference, placement }) => ({
              right: Math.min(popper.width, reference.width) - 24,
            }),
          },
        },
      ]}
```

<img width="353" alt="image" src="https://user-images.githubusercontent.com/339308/193010794-b01b38ec-d8e3-4b2e-a2b7-cd19752adb18.png">

**Comment by florensvb** [comment]
Thanks a lot for this. I ended up modifying the code because the positioning was jumpy when breaking between mobile and desktop layouts. Using the below, the arrow stays pinned to the left (with a `32px` padding) of the input field and does not jump around between different layouts:

```javascript
const popperModifiers = [
  {
    name: 'arrow',
    options: {
      padding: ({ popper }) => ({
        right: popper.width - 32,
      }),
    },
  },
]
```

**Comment by tobiasthorin** [comment]
Just to add to the above comments, the underlying reason the arrow ends up at the edge in the first place is because it strives to be as close to the center of the INPUT (or clicked element) as possible. So the behavior is correct, it just looks janky because of the border radius, which @kingston has already adressed.

**Comment by Ugoku** [comment]
This appears to be fixed in v5.0.0

**Comment by yogeshthakare4002** [comment]
this issue is not closed yet , and popperModifiers dosen't work, i am using   "react-datepicker": "^7.3.0", this version but still facing the same issue
